aws: fixes ELB creation with multiple subnets per AZ

simonswine opened the pull request:
Only takes the first available subnet in a AZ, ignore other subnets
and log warning about this.

Removes AWS region comparison for subnet AZs. A VPC is only in a single
AWS region.

Fixes #12381

simonswine commented:
Hi everyone,

as far as I can see it the issue here is nothing related to this PR. I see other PR with the same test failing:

> k8s.io/kubernetes/test/integration / integration / TestPersistentVolumeClaimBinder

Can you confirm that?

brendandburns commented on pkg/cloudprovider/providers/aws/aws.go:
```diff
@@ -1590,6 +1590,37 @@ func (s *AWSCloud) createTags(request *ec2.CreateTagsInput) (*ec2.CreateTagsOutp
+		return nil, err
+	}
+
+	availabilityZones := make(map[string]bool)
```
please use `pkg/util/set/String` for string sets.

brendandburns commented:
one minor nit.

simonswine committed:
aws: fixes ELB creation with multiple subnets per AZ

Only takes the first available subnet in a AZ, ignore other subnets
and log warning about this.

Removes AWS region comparison for subnet AZs. A VPC is only in a single
AWS region.

Fixes #12381

simonswine commented:
@brendandburns The az check is using the string sets package now. Sorry for the delay. You can blame Oktoberfest for that :laughing:

brendandburns commented:
LGTM, thanks for the PR.
